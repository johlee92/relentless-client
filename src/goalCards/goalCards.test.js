import React from 'react';
import ReactDOM from 'react-dom';
// because using Enzyme to test
// need to make sure the setupTest.js file links to adapter
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import GoalCards from './goalCards';
import { BrowserRouter } from 'react-router-dom';

describe(`GoalCards component`, () => {
    const props = {
        className: 'test-class-name',
        goal:  {
            complete: 'test'
        },
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter> 
                <GoalCards {...props}/>
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    
    it('renders the component from props', () => {
        const wrapper = shallow(<GoalCards {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
