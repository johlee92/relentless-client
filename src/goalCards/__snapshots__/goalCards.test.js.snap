// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalCards component renders the component from props 1`] = `
<div
  className="goal-cards"
>
  <div>
    <button
      onClick={[Function]}
    >
      Status: 
      Complete
    </button>
    <button
      onClick={[Function]}
    >
      Edit
    </button>
  </div>
</div>
`;
